framework/packages/abstract-client/src/publisher.rs

```diff
@@ -3,106 +3,52 @@
-        mut account_builder: AccountBuilder<'a, Chain>,
-        namespace: Namespace,
-    ) -> Self {
-        account_builder.namespace(namespace);
-        Self { account_builder }
-    }
-
-    /// Username for the account
-    /// Defaults to "Default Abstract Account"
-    pub fn name(&mut self, name: impl Into<String>) -> &mut Self {
-        self.account_builder.name(name);
-        self
-    }
-
-    /// Description for the account
-    pub fn description(&mut self, description: impl Into<String>) -> &mut Self {
-        self.account_builder.description(description);
-        self
-    }
-
-    /// http(s) or ipfs link for the account
-    pub fn link(&mut self, link: impl Into<String>) -> &mut Self {
-        self.account_builder.link(link);
-        self
-    }
-
-    /// Overwrite the configured namespace
-    pub fn namespace(&mut self, namespace: Namespace) -> &mut Self {
-        self.account_builder.namespace(namespace);
-        self
-    }
-
-    /// Governance of the account.
-    /// Defaults to the [`GovernanceDetails::Monarchy`] variant, owned by the sender
-    pub fn ownership(&mut self, ownership: GovernanceDetails<String>) -> &mut Self {
-        self.account_builder.ownership(ownership);
-        self
-    }
-
-    /// Install modules on a new sub-account instead of current account.
-    /// Defaults to `true`
-    pub fn install_on_sub_account(&mut self, value: bool) -> &mut Self {
-        self.account_builder.install_on_sub_account(value);
-        self
-    }
-
-    /// Builds the [`Publisher`].
-    /// Creates an account if the namespace is not already owned.
-    pub fn build(&self) -> AbstractClientResult<Publisher<Chain>> {
-        let account = self.account_builder.build()?;
-        Ok(Publisher { account })
-    }
-}
-
 /// A Publisher represents an account that owns a namespace with the goal of publishing modules to the on-chain module-store.
 pub struct Publisher<Chain: CwEnv> {
     account: Account<Chain>,
 }
 
 impl<Chain: CwEnv> Publisher<Chain> {
+    /// New publisher from account. We check that the account has an associated namespace.
+    /// If you create a publisher from an account without a namespace, use [`Self::new_with_namespace`] to claim it
+    pub fn new(account: &Account<Chain>) -> AbstractClientResult<Self> {
+        let namespace = account
+            .infrastructure()?
+            .registry
+            .namespaces(vec![account.id()?])?;
+
+        if namespace.namespaces.is_empty() {
+            return Err(AbstractClientError::NoNamespace {
+                account: account.id()?,
+            });
+        }
+
+        Ok(Self {
+            account: account.clone(),
+        })
+    }
+
+    /// New publisher from account and namespace
+    /// Claim the namespace from the account
+    pub fn new_with_namespace(
+        account: Account<Chain>,
+        namespace: &str,
+    ) -> AbstractClientResult<Self> {
+        account.claim_namespace(namespace)?;
```

Comment:
What if it already have claimed namespace, will it error?

Reply:
I would prefer not, to have more returnable script

Reply:
What do you think now ?

Reply:
https://github.com/AbstractSDK/abstract/pull/503/commits/853a4ba5267e3d83e7ddab451f32b4cdc076f29e#diff-0908c6e723f19d3802d488bc582b29bb0da58a0330aaa450fcfe6e70e5bcee13R49